language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/project/ModuleVisitor.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://wso2.com) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ballerinalang.langserver.extensions.ballerina.project;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import io.ballerina.compiler.syntax.tree.FunctionDefinitionNode;
+import io.ballerina.compiler.syntax.tree.Node;
+import io.ballerina.compiler.syntax.tree.NodeVisitor;
+import io.ballerina.compiler.syntax.tree.ServiceDeclarationNode;
+import io.ballerina.compiler.syntax.tree.SyntaxKind;
+import io.ballerina.compiler.syntax.tree.TypeDescriptorNode;
+import io.ballerina.tools.text.LinePosition;
+
+import java.util.Optional;
+
+/**
+ * Visitor class for module visitor.
+ */
+public class ModuleVisitor extends NodeVisitor {
```

Comment:
Instead of using the visitor to capture the symbols, what's the possibility of using `https://github.com/ballerina-platform/ballerina-lang/blob/master/compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/SemanticModel.java#L60` API?

Reply:
Reported current limitations in https://github.com/ballerina-platform/ballerina-lang/issues/27493